google/datalab/bigquery/commands/_bigquery.py

```diff
@@ -316,6 +316,73 @@ def _create_load_subparser(parser):
   return load_parser
 
 
+def _create_pipeline_subparser(parser):
+  pipeline_parser = parser.subcommand('pipeline', 'Load from GCS, query, and extract data from GCS '
+                                                  'into a BigQuery table. If creating a new table, '
+                                                  'a schema should be specified in YAML or JSON in '
+                                                  'the cell body, otherwise the schema is inferred '
+                                                  'from existing table.')
+
+  # common arguments
+  pipeline_parser.add_argument('-b', '--billing', type=int, help='BigQuery billing tier')
+  # TODO(rajivpb): I've naively made this a common param. Any objections?
+  pipeline_parser.add_argument('-nc', '--nocache', help='Don\'t use previously cached results',
+                               action='store_true')
+
+  # load arguments
+  pipeline_parser.add_argument(
+      '-lm', '--load_mode', help='One of create (default), append or overwrite for loading data',
```

Comment:
I think we want the concept of an input data source which can either be an existing table or a set of files in GCS. This is better represented as a section in the YAML spec rather than a set of arguments. Example:

    input:
      type: table|gcs
      path: gs://... (with placeholders for variables)
      name: table-name-with-placeholders-for-date-or-some-other-variable

Also, I don't know if we should _actually_ load data into a table. We could probably just use a federated data source and query directly against the GCS data. Otherwise it raises questions of who/what deletes the table after the pipeline has run.

Reply:
Agreed on this being in the YAML. I already changed this in https://github.com/googledatalab/pydatalab/pull/494

I have been heretofore thinking of a user experience that specified a pipeline in a way that was conceptually familiar to a Datalab user - i.e. having it looked like it is composed of bq.load, bq.execute, and bq.extract. But I see the simplicity and compactness in your proposed user experience, so let me flesh this out some more with the other required parameters for such an experience (and also respond more fully to your comments in https://github.com/googledatalab/pydatalab/pull/494). 

Perhaps we can leave the composition of individual bq tasks to %%pipeline.